neqo-transport/src/connection.rs

```diff
@@ -1478,6 +1478,11 @@ impl Connection {
         }
     }
 
+    // TODO: this should be #[cfg(test)], but it is not acceptable from a different crate.
+    pub fn handle_max_data_test(&mut self, maximum_data: u64) {
+        self.handle_max_data(maximum_data);
+    }
```

Comment:
We need a different way to handle this.

What should be happening in your test is that the client hits the flow control limit and sends a STREAM_DATA_BLOCKED frame.  The server should respond by increasing the flow control limits.  That should suffice for your test, meaning that you don't need this function at all.

Reply:
I will leave it in this PR and change it #599.

Reply:
#599 has been updated.